ignite/services/plugin/plugin.go

```diff
@@ -0,0 +1,338 @@
+			p.build()
+		}
+	}
+	if p.Error != nil {
+		return
+	}
+	// pluginMap is the map of plugins we can dispense.
+	pluginMap := map[string]hplugin.Plugin{
+		p.binaryName: &InterfacePlugin{},
+	}
+	// Create an hclog.Logger
+	logger := hclog.New(&hclog.LoggerOptions{
+		Name:   fmt.Sprintf("plugin %s", p.Name),
+		Output: os.Stdout,
+	})
+	// We're a host! Start by launching the plugin process.
+	client := hplugin.NewClient(&hplugin.ClientConfig{
+		HandshakeConfig: handshakeConfig,
+		Plugins:         pluginMap,
+		Logger:          logger,
+		Cmd:             exec.Command(p.binaryPath()),
+		SyncStderr:      os.Stderr,
+		SyncStdout:      os.Stdout,
+	})
+	// defer client.Kill()
+
+	// Connect via RPC
+	rpcClient, err := client.Client()
+	if err != nil {
+		p.Error = errors.Wrapf(err, "connecting")
+		return
+	}
+
+	// Request the plugin
+	raw, err := rpcClient.Dispense(p.binaryName)
+	if err != nil {
+		p.Error = errors.Wrapf(err, "dispensing")
+		return
+	}
+
+	// We should have an Interface now! This feels like a normal interface
+	// implementation but is in fact over an RPC connection.
+	p.Interface = raw.(Interface)
+}
+
+// fetch clones the plugin repository at the expected reference.
+func (p *Plugin) fetch() {
+	if p.isLocal() {
+		return
+	}
+	if p.Error != nil {
+		return
+	}
+	defer clispinner.New().SetText(fmt.Sprintf("Fetching plugin %q...", p.cloneURL)).Stop()
+
+	var err error
+	if p.reference == "" {
+		// No reference provided, just clone
+		_, err = git.PlainClone(p.cloneDir, false, &git.CloneOptions{
+			URL: p.cloneURL,
+		})
+	} else {
+		// Reference provided, clone using tag or branch reference, one of the two
+		// should work. SHA-1 aren't supported.
+		for _, ref := range []plumbing.ReferenceName{
+			plumbing.NewTagReferenceName(p.reference),
+			plumbing.NewBranchReferenceName(p.reference),
+		} {
+			_, err = git.PlainClone(p.cloneDir, false, &git.CloneOptions{
+				URL:           p.cloneURL,
+				ReferenceName: ref,
+				// Try to limit number of commits but this option doesn't seem to work well
+				Depth: 1,
+			})
+			if err == nil {
+				break
+			}
+		}
+	}
+	if err != nil {
+		p.Error = errors.Wrapf(err, "cloning %q", p.cloneURL)
+	}
+}
+
+// build compiles the plugin binary.
+func (p *Plugin) build() {
+	if p.Error != nil {
+		return
+	}
+	defer clispinner.New().SetText(fmt.Sprintf("Building plugin %q...", p.Name)).Stop()
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		p.Error = errors.Wrapf(err, "build")
+		return
+	}
+	err = os.Chdir(p.srcPath)
+	if err != nil {
+		p.Error = errors.Wrapf(err, "build")
+		return
+	}
+	// Back to previous dir
+	defer os.Chdir(cwd)
+
+	run := func(name string, args ...string) {
+		cmd := exec.Command(name, args...)
+		var b bytes.Buffer
+		cmd.Stderr = &b
+		if err := cmd.Run(); err != nil {
+			p.Error = errors.Wrapf(err, "build: %s: %s", cmd, b.String())
+		}
+	}
+
+	run("go", "mod", "tidy")
+	if p.Error != nil {
+		return
+	}
+	run("go", "build", "-o", p.binaryName)
```

Comment:
These could be replaced by `gocmd.ModTidy` and `gocmd.BuildPath`

Reply:
thanks, that simplifies a lot (no need to deal with working dir!)
 aabb88ae